R/init.R

```diff
@@ -228,7 +245,10 @@ init <- function(data,
       if (!is.null(landing_module)) {
         do.call(landing_module$server, c(list(id = "landing_module_shiny_id"), landing_module$server_args))
       }
-      srv_teal_with_splash(id = id, data = data, modules = modules, filter = deep_copy_filter(filter))
+      srv_teal_with_splash(
+        id = id,
+        data = data, modules = modules, filter = deep_copy_filter(filter), lockfile_task = lockfile_task
+      )
```

Comment:
I think I raised this somewhere. Is it possible to move the functionality into a function so that we won't change arguments of any of the functions that is being called?

Reply:
@pawelru yes you did, and then we had a couple more iterations on this one. So I will write it again. No worries. The actual need to calculate it in `teal::init` and then to pass to `srv_teal_with_splash` is because if you calculate that in `teal::init` then .lockfile calculation is done once and shared across the app. If calculations are done in `srv_teal` then each session has separate independent .lockfile calculations

Reply:
This is actually quite relevant. Some of the modules has `require()` statement which essentially means that you can end up having different session out of the same app. On that regard, putting this inside `srv_teal_with_splash` feels more correct to me.

Reply:
yeah, so you are far behind what was discussed in here. renv does not store the session, but stores what is in project files. and those are always the same

Reply:
so no need to recalculate the renv more than once as it's always gonna be the same

Reply:
Yes you are right!